spec.html

```diff
@@ -39097,6 +38930,256 @@
+        1. Perform ? _m_.ModuleDeclarationInstantiation().
+        1. Assert: All dependencies of _m_ have been transitively resolved and _m_ is ready for evaluation.
+        1. Return ? _m_.ModuleEvaluation().
+      </emu-alg>
+      <emu-note>
+        <p>The implementation might parse a _sourceText_ as a |Module|, analyze it for Early Error conditions, and instantiate it prior to the execution of the TopLevelModuleEvaluationJob for that _sourceText_. It might also resolve, pre-parse and pre-analyze, and pre-instantiate module dependencies of _sourceText_. However, the reporting of any errors detected by these actions will be deferred until the TopLevelModuleEvaluationJob is actually executed.</p>
+      </emu-note>
+    </emu-annex>
+
+    <emu-annex id="sec-example-module-resolution">
+      <h1>Module Resolution</h1>
+      <p>This implementation uses the [[HostDefined]] field of Source Text Module Records to store the filename from which a module was retrieved. It also maintains a List of ModuleMap Records, known as the module map. This helps enforce the idempotency requirements of HostResolveImportedModule. ModuleMap Records have the fields described here:</p>
+
+      <emu-table id="table-example-module-map-record" caption="ModuleMap Record Fields">
+        <table>
+          <thead>
+          <tr>
+            <th>
+              Field Name
+            </th>
+            <th>
+              Value
+            </th>
+            <th>
+              Meaning
+            </th>
+          </tr>
+          </thead>
+          <tbody>
+          <tr>
+            <td>
+              [[Filename]]
+            </td>
+            <td>
+              A String
+            </td>
+            <td>
+              The filename from which this module was created.
+            </td>
+          </tr>
+          <tr>
+            <td>
+              [[Module]]
+            </td>
+            <td>
+              A Source Text Module Record
+            </td>
+            <td>
+              The module record created for this module.
+            </td>
+          </tr>
+          </tbody>
+        </table>
+      </emu-table>
+
+      <p>This implementation's HostResolveImportedModule abstract operation performs the following steps, given the arguments _referencingModule_ and _specifier_:</p>
+      <emu-alg>
+        1. Let _referencingFilename_ be _referencingModule_.[[HostDefined]].
+        1. Let _filename_ be the result of resolving _specifier_ against _referencingFilename_ as a filename, according to operating system conventions.
+        1. If the module map contains a ModuleMap Record with [[Filename]] equal to _filename_, return that Record's [[Module]] field.
+        1. Let _sourceText_ be the result of using the operating system's mechanisms to read the source text of the file located at _filename_. If this fails (e.g. because the file does not exist), throw a *TypeError*.
+        1. Let _moduleRecord_ be ? ParseModule(_filename_).
```

Comment:
`ParseModule` is confusing here since it is not the abstract operation defined in the specification.